### PR TITLE
Rendered Components - Appbar, SideBar, Avatar, Splitter, Mention.

### PR DESCRIPTION
**Pull request by SenthilarasuBalu**
I have rendered  the  following components and Updated the TS WebMail showcase application.
* Appbar.
* Avatar.
* Sidebar.
*  Mention.
*  Splitter.

BeforeImplementingAppbar_and_AvatarComponent
BeforeImplementingAppbar_and_AvatarComponent

AfterImplementingAppbar_and_Avatar_Component.
AfterImplementingAppbar_and_Avatar Component

BeforeImplementingSidebarComponent
BeforeImplementingSidebarComponent

AfterAdditionOfSidebarComponentInDesktopView.
AfterAdditionOfSidebarComponentInDesktopView

SidebarComponent in Tabletview.
SidebarComponent in Tabletview

SideBarComponentInMobileView.
SideBarComponentInMobileView

BeforeSplitterComponentImplementation.
BeforeSplitterComponent

AfterAddingSplitterComponentDesktopView.
AfterAddingSplitterComponentDesktopView

SplitterComponent In Tabletview.
SplitterComponent In Tabletview

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,69 +1,73 @@
 <!DOCTYPE html>
 <html lang="en">
+
 <head>
     <title>Essential JS 2 for TypeScript - Webmail</title>
     <meta charset="utf-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no" />
-    <meta name="description" content="Essential JS 2 for TypeScript - Webmail - This demo explores a mailbox user interface which was inspired by the Outlook Office 365 application" />
+    <meta name="description"
```
Comment (NavinVinayagam) [inline index.html]: Avoid alignment unnecessary changes

File: package.json [hunk 2]
```diff
@@ -10,28 +10,29 @@
     "hasher": "^1.2.0"
   },
   "devDependencies": {
-    "@types/chai": "^3.4.28",
-    "@types/hasher": "0.0.27",
-    "@types/crossroads": "0.0.28",
-    "@types/jasmine": "2.8.9",
-    "@types/jasmine-ajax": "^3.1.27",
-    "@types/requirejs": "^2.1.26",
     "@syncfusion/ej2-base": "*",
```
Comment (NavinVinayagam) [inline package.json]: Why so many unnecessary changes, Add only the required packages and don't change the existing order

File: index.html [hunk 3]
```diff
@@ -1,69 +1,73 @@
...
-                    <div style="margin-right: 44px; pointer-events: none;">
-                        <button id='btnLoginName' class="btn-profile" style="min-width: 88px;font-weight: normal"></button>
-                    </div> 
+            </div>
+            <div id="profile-div" class="header-right-pane "> 
+                <div style=" pointer-events: none;">
+                    <button id="btnLoginName" class="btn-profile e-control e-btn e-lib btn-shadow-hide" style="min-width: 88px;font-weight: normal">Andrew Fuller</button>
                 </div>
+                <div style="pointer-events: none;margin-right: 5px;">
+                    <img class="e-avatar e-avatar-large e-avatar-circle" style="height: auto; width: auto;" src="./styles/images/1.png">
```
Comment (NavinVinayagam) [inline index.html]: Use SPAN tag with background image instead of IMG tag

File: src/home/home.html [hunk 4]
```diff
@@ -1,32 +1,6 @@
...
-                </div>
-            </div>
-            <div class="folder-pane customScrollBar">
-                <div id='tree'></div>
-            </div>
-        </div>
-    </div>
-    <div class="headerContent" style="z-index: 112; height: 100%; background-color: #f4f4f4; width: 330px;right:0; position: absolute">
+    <div class="headerContent"
+        style="z-index: 112; height: 100%; background-color: #f4f4f4; width: 330px;right:0; position: absolute">
```
Comment (NavinVinayagam) [inline src/home/home.html]: avoid unnecessary alignment change

File: src/home/home.html [hunk 5]
```diff
@@ -112,42 +86,76 @@
         </div>
     </div>
     <div class='content-wrapper' role='main'>
+        <div id="sideBar" class='sidebar' style='height:100%' role='complementary'>
```
Comment (NavinVinayagam) [inline src/home/home.html:88]: What is the need for moving the sidebar element?

File: src/home/home.html [hunk 6]
```diff
@@ -112,42 +86,76 @@
...
+            <div class='maincontent_pane'
+                style='background-color:#fff; width:100%;height:calc(100% - 40px);color:white'>
+
+                <div id="list-pane-div" class='message-pane customScrollBar' style='overflow: auto'>
                     <div class="filter-section">
                         <div style='float: left; margin-top: 5px;'>
                             <span id="spanFilterText" style='font-size: 22px;color:#333333'>Inbox</span>
                         </div>
-                        <div style='right:25px; position: absolute; top:10px'>
+                        <div style='right:25px; position: absolute; top:15px'>
```
Comment (NavinVinayagam) [inline src/home/home.html]: Why top value is increased

File: src/home/home.ts [hunk 7]
```diff
@@ -86,23 +92,38 @@ window.home = (): void => {
         document.getElementById('reading-pane-popup').innerHTML = value.toString();
         window.readingpane();
     });
+
+    let appObject: AppBar = new AppBar({
+        colorMode: 'Dark'
+
```
Comment (NavinVinayagam) [inline src/home/home.ts]: remove unwanted empty lines

File: src/home/home.ts [hunk 8]
```diff
@@ -1076,25 +1103,27 @@ function showNewMailPopup(option: string): void {
     document.getElementById('mailarea').appendChild(document.getElementById('newmailContent'));
     (document.getElementsByClassName('tb-item-new-mail')[0] as HTMLElement).style.display = 'none';
     (document.getElementsByClassName('tb-item-mark-read')[0] as HTMLElement).style.display = 'none';
+    document.getElementById('toolbar_align').style.display = 'none';
```
Comment (NavinVinayagam) [inline src/home/home.ts:1109]: What is the need for this element

File: src/home/newmail.ts [hunk 9]
```diff
@@ -103,7 +110,7 @@ function createMailForm(): void {
         ],
         height: '100%',
         width: '100%',
-        overflowMode: 'Popup'
+        overflowMode: 'Scrollable'
```
Comment (NavinVinayagam) [inline src/home/newmail.ts:111]: WHy mode is changed

File: src/home/newmail.html [hunk 10]
```diff
@@ -14,10 +14,11 @@
         <input type='text' id='autoCc' />
     </div>
     <div class="mail-subject e-input-group">
-        <input type='text' id='txtSubject' style="width: 100%" placeholder="Enter Subject Here" class = "e-input"/>
+        <input type='text' id='txtSubject' style="width: 100%" placeholder="Enter Subject Here" class="e-input" />
     </div>
     <div>
-        <div id='mailContentMessage' class="customScrollBar" contenteditable="true"></div>
+        <div id="mentionElement" style="border: 1px solid #D7D7D7; border-radius: 4px; padding: 8px; font-size: 14px;  height: 100%; width: 100%;">
```
Comment (NavinVinayagam) [inline src/home/newmail.html]: Why mailContentMessage element is removed and mentionElement is added?
Why lot's of inline styles are added?

File: styles/css/outlook.css [hunk 11]
```diff
@@ -1684,4 +1696,29 @@ html, body {
...
+    padding: 0px;
+}
+.header-right-pane
+{
+    height: 50px;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+}
+.e-splitter .e-split-bar.e-split-bar-horizontal, .e-splitter .e-split-bar.e-split-bar-horizontal.e-split-bar-hover,
```
Comment (NavinVinayagam) [inline styles/css/outlook.css]: Add comment line regaridng the purpose of overwriting the CSS

**Review by NavinVinayagam — CHANGES_REQUESTED** [review]

File: styles/css/outlook.css [hunk 12]
```diff
@@ -1450,19 +1443,29 @@ html, body {
...
 @media screen and (max-width:1090px) {
+    
     #toolbar_align {
         display: none;
     }
     #toolbar_mobile {
         display: block;
     }
-    .header.navbar {
+    #toolbar-left-head
```
Comment (NavinVinayagam) [inline styles/css/outlook.css]: use proper format

File: src/home/home.html (hunk 5 above)
Comment (SenthilarasuBalu) [reply src/home/home.html:88]: Wrapper div is needed for the Sidebar to work and for avoiding UI Glitch while (Push).

File: src/home/newmail.ts (hunk 9 above)
Comment (SenthilarasuBalu) [reply src/home/newmail.ts:111]: While re-sizing Splitter Component, Toolbar menus overflow outside the container. To avoid this situation, Overflow mode has been changed.

File: styles/css/outlook.css (hunk 11 above)
Comment (SenthilarasuBalu) [reply styles/css/outlook.css]: Comment Lines has been added for  overwriting the CSS.

File: index.html (hunk 1 above)
Comment (SenthilarasuBalu) [reply index.html]: Unnecessary Changes of alignment has been modified.

File: package.json (hunk 2 above)
Comment (SenthilarasuBalu) [reply package.json]: Added  the required packages, Without changing the existing order.

File: index.html (hunk 3 above)
Comment (SenthilarasuBalu) [reply index.html]: SPAN tag has been used.

File: src/home/home.html (hunk 4 above)
Comment (SenthilarasuBalu) [reply src/home/home.html]: Unnecessary Changes of alignment has been modified.

File: src/home/home.ts (hunk 7 above)
Comment (SenthilarasuBalu) [reply src/home/home.ts]: Empty lines has been removed.

File: styles/css/outlook.css (hunk 12 above)
Comment (SenthilarasuBalu) [reply styles/css/outlook.css]: Changes done.

File: src/home/newmail.html (hunk 10 above)
Comment (SenthilarasuBalu) [reply src/home/newmail.html]: Format has been restructured.

File: src/home/home.ts (hunk 8 above)
Comment (SenthilarasuBalu) [reply src/home/home.ts:1109]: The element is needed for fixing the alignment issue With out affecting the Splitter component.

File: index.html [hunk 13]
```diff
@@ -24,46 +24,46 @@
...
                 </div>
+                <div style="margin-right: 5px; pointer-events: none;">
+                    <span class="e-avatar e-avatar-large e-avatar-circle" style="background-image: url('./styles/images/1.png');height: 40px; width: 40px;"></span>
+                </div>  
             </div>
         </div>
-        <div id="content-area" style="height: 100%">
-        </div>
+
+    <div id="content-area" style="height: 100%">
```
Comment (NavinVinayagam) [inline index.html]: Check the alignment here

File: src/home/home.html [hunk 14]
```diff
@@ -112,15 +85,43 @@
...
             <div class="e-roottools" style='background-color:#fff; width:100%;color:white'>
                 <div id='toolbar_align'>
                 </div>
                 <div id='toolbar_mobile'>
                 </div>
             </div>
-            <div class='maincontent_pane' style='background-color:#fff; width:100%;height:calc(100% - 40px);color:white'>
-                <div class='message-pane customScrollBar' style='overflow: auto'>
+            <div class='maincontent_pane' style='background-color:#f4f9fd; width:100%;height:100%;color:white'>
+            <div id="splitter"></div>
```
Comment (NavinVinayagam) [inline src/home/home.html]: add a tab space

File: src/home/home.ts [hunk 15]
```diff
@@ -1,16 +1,17 @@
-/**
+/**mboboxmbobox
```
Comment (NavinVinayagam) [inline src/home/home.ts]: Remove this, and ensure the changes before committing

File: src/home/home.ts [hunk 16]
```diff
@@ -1103,22 +1125,56 @@ function onWindowResize(evt: Event): void {
     }
     if (window.innerWidth < 1090) {
         contentArea.className = 'row content sidebar-hide';
+        if (messagePane.classList.contains('pane-spacer')){
```
Comment (NavinVinayagam) [inline src/home/home.ts]: If condition not required here

File: src/home/home.ts [hunk 17]
```diff
@@ -1103,22 +1125,56 @@ function onWindowResize(evt: Event): void {
...
+            messagePane.classList.remove("pane-spacer");
+        }
+        defaultSidebar.hide();
+        defaultSidebar.type='Over';
     } else {
-        hideSideBar();
+        if (!isNewMailClick){
```
Comment (NavinVinayagam) [inline src/home/home.ts]: Simplify this.
Refer the line: https://github.com/essential-studio/ej2-button-components/blob/development/src/speed-dial/speed-dial.ts#L899

File: src/home/home.ts [hunk 18]
```diff
@@ -1446,4 +1502,8 @@ function openPopup(): void {
     setTimeout(() => { hidePopup(); }, 2000);
 }
 
-setTimeout(openPopup, 3000);
\ No newline at end of file
+setTimeout(openPopup, 3000);
+function removeSpacer() {
```
Comment (NavinVinayagam) [inline src/home/home.ts]: updateNewMailClick

File: src/home/home.ts [hunk 19]
```diff
@@ -1446,4 +1502,8 @@ function openPopup(): void {
...
+    isNewMailClick=false;
+    document.getElementById('list-pane-div').classList.remove("pane-spacer");
```
Comment (NavinVinayagam) [inline src/home/home.ts]: Use the class name as msg-top-margin

File: src/home/home.ts [hunk 20]
```diff
@@ -86,6 +90,17 @@ window.home = (): void => {
         document.getElementById('reading-pane-popup').innerHTML = value.toString();
         window.readingpane();
     });
+    let appObject: AppBar = new AppBar({
+        colorMode: 'Dark'
+    })
+    appObject.appendTo("#appbar");
+    defaultSidebar = new Sidebar({
+        width: "280px",
+        type: "Push",
```
Comment (NavinVinayagam) [inline src/home/home.ts:97]: Use the sidebar backdrop and remove the overlay element added manually

**Review by NavinVinayagam — CHANGES_REQUESTED** [review]

File: index.html (hunk 13 above)
Comment (SenthilarasuBalu) [reply index.html]: Aligned.

File: src/home/home.html (hunk 14 above)
Comment (SenthilarasuBalu) [reply src/home/home.html]: Tab Space Added.

File: src/home/home.ts (hunk 15 above)
Comment (SenthilarasuBalu) [reply src/home/home.ts]: Corrected it.

File: src/home/home.ts (hunk 16 above)
Comment (SenthilarasuBalu) [reply src/home/home.ts]: if condition removed.

File: src/home/home.ts (hunk 17 above)
Comment (SenthilarasuBalu) [reply src/home/home.ts]: Simplified.

File: src/home/home.ts (hunk 18 above)
Comment (SenthilarasuBalu) [reply src/home/home.ts]: Function Name changed.

File: src/home/home.ts (hunk 19 above)
Comment (SenthilarasuBalu) [reply src/home/home.ts]: Class name modified.

File: src/home/home.ts (hunk 20 above)
Comment (SenthilarasuBalu) [reply src/home/home.ts:97]: Used the sidebar backdrop .

**Review by NavinVinayagam — APPROVED** [review]

File: src/home/home.ts [hunk 21]
```diff
@@ -416,6 +443,7 @@ export function showSelectedMessage(): void {
     readingPane.className = readingPane.className.replace(' new-mail', '');
     (document.getElementsByClassName('tb-item-new-mail')[0] as HTMLElement).style.display = 'inline-flex';
     (document.getElementsByClassName('tb-item-mark-read')[0] as HTMLElement).style.display = 'none';
+    document.getElementById('toolbar_align').style.display = '';
```
Comment (NavinVinayagam) [inline src/home/home.ts:446]: Why display is set to empty?

**Review by NavinVinayagam — CHANGES_REQUESTED** [review]

File: styles/styles.scss [hunk 22]
```diff
@@ -33,3 +33,7 @@
 @import 'ej2-dropdowns/styles/multi-select/all.scss';
 @import 'ej2-popups/styles/tooltip/fabric-definition.scss';
 @import 'ej2-popups/styles/tooltip/all.scss';
+@import 'ej2-base/styles/fabric.scss';
+@import 'ej2-navigations/styles/fabric.scss';
```
Comment (NavinVinayagam) [inline styles/styles.scss]: Can we simplify the these imports to more specific packages?

**Review by NavinVinayagam — CHANGES_REQUESTED** [review]

File: styles/styles.scss (hunk 22 above)
Comment (SenthilarasuBalu) [reply styles/styles.scss]: Imports Changes Done.

File: src/home/home.ts (hunk 21 above)
Comment (SenthilarasuBalu) [reply src/home/home.ts:446]: It behaves according to elements nature.

**Review by NavinVinayagam — APPROVED** [review]

**Review by syncsiva — APPROVED** [review]
